### PR TITLE
Setup core development environment

### PR DESCRIPTION
**Pull request by enioluwa23**
This PR sets up the Python environment for developing the main components of the project. DevOps is also set up:
- Mypy for type checking
- Flake8 and PyLint for linting
- Bandit for security testing
- Pytest for tests
- Tox for putting it all together

A WIP version of contributing instructions is also written here. This partly addresses #3 but does not close it.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+- NIX* environment (Linux, WSL, macOS terminal)
+- Python (3.5 and above)
+- Pip
+
+Keep in mind that we enforce the use of python3 type hints as much as possible, to avoid obscurity.
+
+Optionally, though strongly recommended, is to install Python's virtual environment (venv) module which will help with package management and dependency isolation. The rest of this tutorial assumes you have chosen to use a venv. 
+
+Follow these steps to finish setting up:
+1. Spin up a venv in the core folder, and activate it using any equivalent shell command to bash's `source ./bin/activate`.
```
Comment (ryanhassing) [inline CONTRIBUTING.md]: This step isn't clear to me.

Comment (enioluwa23) [reply CONTRIBUTING.md]: Right, thanks for the catch. Will add instructions on how to start up a venv.

Comment (enioluwa23) [reply CONTRIBUTING.md]: @ryanhassing take another look.

Comment (ryanhassing) [reply CONTRIBUTING.md]: 'spin up a venv in the core folder' - still not obvious to me what this means. Maybe just write the CLI steps needed:
`cd core`
`venv`
or whatever steps are necessary.

Comment (enioluwa23) [reply CONTRIBUTING.md]: As the tag on this code block says, that is the outdated version that you first read. Comment on the changes from the latest commit.

**Pull request merged**